src/sentry/auth/helper.py

```diff
@@ -424,7 +425,7 @@ def _logged_in_user(self) -> Optional[User]:
     @property
     def _app_user(self) -> Optional[User]:
         """The user, if they are represented persistently in our app."""
-        return self.user if isinstance(self.user, User) else None
+        return self.user if isinstance(self.user, BaseUser) else None
```

Comment:
Hmm, feeling this also is probably not the best spot for BaseUser.  I'll revert.